luminoth/train.py

```diff
@@ -53,9 +54,16 @@ def run(custom_config, model_type, override_params, target='',
             config['dataset']['type']
         except KeyError:
             raise KeyError('dataset.type should be set on the custom config.')
-        dataset_class = get_dataset_fn(config.dataset.type)
-        dataset = dataset_class(config)
-        train_dataset = dataset()
+
+        try:
+            dataset_class = get_dataset_fn(config.dataset.type)
+            dataset = dataset_class(config)
+            train_dataset = dataset()
+        except InvalidDataDirectory as exc:
+            tf.logging.error(
+                "Error while reading dataset, {}".format(exc)
+            )
+            return
```

Comment:
I would change `tf.logging.error` to `raise InvalidDataDirectory` (error string being the same) because currently we're exiting with exit code `0`, which doesn't seem like the right thing to do (as, e.g. I think google cloud ml jobs are marked as successful if the exit code is `0`).

That said, there may be some advantage to using `tf.logging.error` unknown to me. In that case, it may be a good idea to do something along the lines of `tf.logging.error`ing a constant message and then re-raising the `InvalidDataDirectory` exception.

cc @vierja to see what he thinks about this.

Reply:
I just added a task to catch exceptions thrown when someone screws up with the config to avoid a big exception traceback in the cli.

Also, in the transform tool I chose to [manually exit with error](tryolabs/luminoth/blame/master/luminoth/tools/dataset/transform.py#L42), but that was just because the code is simpler and I think that is the only possible exception.

For now I would keep the `tf.logging.error` AND follow @IanTayler's recommendation to raise the exception. We'll then add the train error handling in another PR.